### PR TITLE
Bug/496 bug rohdaten cannot be deleted anymore

### LINKED ISSUE #496: Rohdaten cannot be deleted anymore
The actual dev version (today 2024-07-26 and some versions back, minimum from Feb 2024 onwards) are not working anymore when trying to delete a CMS element from "Rohdaten" tab. The confirmation popup comes up when pressing the delete button, but no action is executed after confirming the popup request. Tested under Win11 and in various browsers (Chrome, FF, Edge).

### PR DESCRIPTION
**Pull request by dceonline**
fixed, see commits

@tone2tone @muratpurc please review and test

### CODE REVIEW

File: contenido/includes/include.con_content_list.php [hunk 1]
```diff
@@ -646,7 +646,7 @@
         $content = $selectedArticle->getContent();
         if ($selectedArticle->isLoaded() && is_array($content)) {
             $result = array_change_key_case($content, CASE_UPPER);
-            $result = $versioning->sortResults($result);
+            $result = $versioning->sortResults((array)$result);
```
Comment (muratpurc) [inline contenido/includes/include.con_content_list.php:649]: The return value of `array_change_key_case()` is already an array.

File: contenido/includes/include.con_content_list.php [hunk 2]
```diff
@@ -751,7 +751,7 @@
 
         if ($selectedArticle->isLoaded() && is_array($content)) {
             $result = array_change_key_case($content, CASE_UPPER);
-            $result = $versioning->sortResults($result);
+            $result = $versioning->sortResults((array)$result);
```
Comment (muratpurc) [inline contenido/includes/include.con_content_list.php:754]: The return value of `array_change_key_case()` is already an array.

File: contenido/includes/include.con_content_list.php (hunk 1 above)
Comment (muratpurc) [inline contenido/includes/include.con_content_list.php:649]: The return value of `array_change_key_case()` is already an array.

File: contenido/includes/include.con_content_list.php [hunk 3]
```diff
@@ -1066,21 +1066,21 @@ function _processCmsTags(
...
             confirmDeletionText = "' . i18n("Are you sure you want to delete this content type from this article?") . '";
         $(function() {
-            $("#con_content_list [data-con-action]").live("click", function() {
+            $("#con_content_list [data-con-action]").on("click", function() {
                 var $element = $(this),
                     action = $element.data("con-action");
                 if (action === "delete_content_type") {
                     var path = pathTemplate.replace("%s", $element.data("con-idcontent"));
                     Con.showConfirmation(confirmDeletionText, function() {
                         console.log("Con.Tiny.setContent", path);
```
Comment (muratpurc) [inline contenido/includes/include.con_content_list.php:1082]: The line with `console.log()` could be removed.

**Review by muratpurc — CHANGES_REQUESTED** [review]

File: contenido/includes/include.con_content_list.php (hunk 1 above)
Comment (dceonline) [reply contenido/includes/include.con_content_list.php:649]: I get an phpstan error in PHPStorm, telling me to cast there. But you are right, there seems to be a bigger thing with that, so i have to check this out. Meanwhile i will remove casting there.

File: contenido/includes/include.con_content_list.php (hunk 3 above)
Comment (dceonline) [reply contenido/includes/include.con_content_list.php:1082]: right, but it was there, so i let it there. We should define how we will handle this console.log entries, and if there some of them, that will stay in code, and some which have to be removed (only used for local testing). Maybe we should only comment them?!

File: contenido/includes/include.con_content_list.php (hunk 1 above)
Comment (muratpurc) [reply contenido/includes/include.con_content_list.php:649]: The IDE also complains at this point and displays following warning:
`Expected parameter of type 'array', 'cApiContent' provided`
But I guess the reason for this is probably something else.

File: contenido/includes/include.con_content_list.php (hunk 3 above)
Comment (muratpurc) [reply contenido/includes/include.con_content_list.php:1082]: This is ok, lets keep it for the moment.

**Review by muratpurc — APPROVED** [review]

**Review by muratpurc — APPROVED** [review]

**Pull request merged**